lib/streamManager.js

```diff
@@ -177,16 +177,20 @@ var exports = module.exports = (function () {
             }));
         },
 
-        flushFunction: function () {
-            end = true;
+        flushFunction: function (cb) {
+            end = cb;
             if (unprocessed === 0) {
-                this.emit('end');
+                return cb();
             }
         },
 
         additionalStreams: function () {
             var options = exports.options;
 
+            additionalFiles = [];
+            unprocessed = 0;
+            end = undefined;
```

Comment:
These are global variables used in gulp-useref that are not reinitialized if you invoke it a 2nd time.  See my comments from yesterday (#2)-- I ran into this problem when a test failed, it left useref in a partially completed state, and subsequent invocations all failed with a timeout because it wasn't reinitializing properly.